src/meshzoo/_helpers.py

```diff
@@ -209,3 +209,65 @@ def _compose_from_faces(corners, faces, n, edge_adjust=None, face_adjust=None):
     cells = np.concatenate(cells)
 
     return vertices, cells
+
+
+def insert_midpoints_edges(points, cells, cell_type):
+    """Collect all unique edges, calculate and return points including
+    midpoints on edges as well as the extended cells array."""
+
+    number_of_points = {"triangle": 3, "tetra": 4, "quad": 4, "hexahedron": 8}
+
+    if cells.shape[1] != number_of_points[cell_type]:
+        raise ValueError("Mismatch of cell type and shape of cells array.")
+
+    if cell_type == "triangle":
+        # k-th edge between cell points no. (i[k], j[k])
+        i = [0, 1, 2]
```

Comment:
Perhaps we can make this one array ij,
```
ij = [
  [0, 1],
  [1, 2],
  [2, 0]
]
```
More readable and saves use the dstack below.

Reply:
I'll give it a try.

Reply:
I think a list is enough, no need for an array here.

Reply:
got it, see 342b37a.